### ISSUE TITLE
Upgrade Coherence

### ISSUE THREAD

**Issue by ysbaddaden** [instedd/surveda#2036]
During the recent upgrades (#1986 and #2019) it proved easier and much quicker to upgrade our fork of Coherence 0.3 rather than trying to upgrade to the latest Coherence.

For obvious reasons, it's not a good idea to continue this way, and before or in parallel, to upgrading to Phoenix 1.5 it would be interesting to try and assess a proper upgrade to the latest `master` branch (sadly there have been no releases for many years, while `master` did get updates) that already supports Phoenix 1.5.

refs #1963
refs #1965

**Comment by ysbaddaden** [comment]
I starting working on thisand followed those steps:

1. I created a new, empty, Phoenix 1.5 project;
2. I installed the latest Coherence `master` with the same feature set;
3. I branched from #2019 (Phoenix 1.4 upgrade);
4. I started applying most changes manually, comparing the new project to the current Surveda;

A significant difference is that the project layout changed in Phoenix 1.3 (I think) but we never made it. This is moving the `/web` folder to `/web/ask_web` and changing the `Web` namespace to `AskWeb`, and moving a bunch of things under that module. This is disturbing, but not much of an issue

A bigger issue is that the controllers have been completely overhauled. Coherence no longer generates whole controllers with actions, but empty controllers 'inheriting' their actions from base controllers, then using responders to somewhat control behavior, such as redirecting after a given action.

Since we made some changes to the original controller behaviors (e.g. optional OAuth with Guisso), we must look at the history of each controller to verify the behavior, and write controller tests to verify the behavior and avoid regressions, while having currently zero automated tests to verify those. We should write some.

Additional notes:

- [ ] We must upgrade the Timex deps to v3.6, but v3.4 broke `Timex.Timezone.resolve` (I'm working on a patch).
